### ISSUE TITLE
ETA 

### ISSUE THREAD

**Issue by dunovank** [noa/uar#1]
Hello, I read the paper that links to this repo and was wondering if you have any estimates on when the code will be available. 

Thanks!

**Comment by noa** [comment]
@dunovank Thanks for your interest. It's currently going through our internal pre-release review, which (unfortunately) has no deadline. However, we're hoping to release another version soon, in the next few weeks.

**Comment by dunovank** [comment]
Thanks for the quick response - and no problem, will check back in a couple weeks.

**Comment by dunovank** [comment]
Hey @noa - I'm curious which of the pretrained SBERT models you used as the text encoder and can't seem to find the specifics in the paper. (Apologies if it's there and I missed it).

**Comment by noa** [comment]
In the original code, it looks like we used:

https://huggingface.co/distilroberta-base

But this shouldn't make too much difference in practice as the training sets are pretty large by "fine-tuning" standards.

**Comment by steven-hh-ding** [comment]
any updates @noa  thanks

**Comment by steven-hh-ding** [comment]
thanks @noa  but I think this is a bit unfair as in the paper it is claimed that reproducibility is provided by the source code. we  were asked by the reviewers to compare with your work. if you have a simple prototye etc released will be helpful.

**Comment by noa** [comment]
@steven-hh-ding We are happy to help reproduce any results from the paper. The method is a simple modification of the NAACL code I shared above, with the following two changes:

1. Replace the CNN layer with BERT, specifically: https://huggingface.co/distilroberta-base 
2. Replace triplet loss with the SupCon loss: https://github.com/google-research/google-research/tree/master/supcon

That being said, I'll see if we can get an unpolished version of our re-implementation (the original had license issues) out soon to help you out.

**Comment by steven-hh-ding** [comment]
Thanks noa for the details. The first step seems straightforward to us but the second step requires dramatic change to the codebase. I don't think it is easily reproducible. All these three things are open source already; could you share a prototype by combining them so we can run?

**Comment by noa** [comment]
@steven-hh-ding Shoot me an email. We can share the original PyTorch code / model files provided you don't re-distribute it.

Re: Step 2, it's actually not that bad: you just need to reshape the input to have a "dummy" view of size 1, and call the contrastive_loss function from the Google repo.

**Comment by dunovank** [comment]
In case anyone is interested, pytorch-metric-learning now includes a torch-compatible implementation of [SupConLoss](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#supconloss)
